debug/log/kubernetes/kubernetes.go

```diff
@@ -2,29 +2,165 @@
 package kubernetes
 
 import (
-	"errors"
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"time"
 
 	"github.com/micro/go-micro/debug/log"
+	"github.com/micro/go-micro/util/kubernetes/client"
 )
 
-type klog struct{}
+type klog struct {
+	client client.Kubernetes
 
-func (k *klog) Read(...log.ReadOption) ([]log.Record, error) {
-	return nil, errors.New("not implemented")
+	log.Options
+}
+
+func (k *klog) Read(options ...log.ReadOption) ([]log.Record, error) {
+	opts := &log.ReadOptions{}
+	for _, o := range options {
+		o(opts)
+	}
+
+	logsToGet, err := k.getMatchingPods()
+	if err != nil {
+		return nil, err
+	}
+	records := []log.Record{}
+
+	for _, l := range logsToGet {
+		logParams := make(map[string]string)
+		if !opts.Since.Equal(time.Time{}) {
+			logParams["sinceSeconds"] = strconv.Itoa(int(time.Since(opts.Since).Seconds()))
+		}
+		if opts.Count != 0 {
+			logParams["tailLines"] = strconv.Itoa(opts.Count)
+		}
+		if opts.Stream == true {
+			logParams["follow"] = "true"
+		}
+		logs, err := k.client.Logs(l, client.AdditionalParams(logParams))
+		if err != nil {
+			return nil, err
+		}
+		defer logs.Close()
+		s := bufio.NewScanner(logs)
+		for s.Scan() {
+			record := k.parse(s.Text())
+			record.Metadata["pod"] = l
+			records = append(records, record)
+		}
+	}
+
+	sort.Sort(byTimestamp(records))
```

Comment:
You can use `sort.Slice` which accepts an inline function. Will stop from needing to define a whole sort type